### PR TITLE
Fix Rust installation in Synapse Docker image

### PR DESCRIPTION
**Pull request by erikjohnston**
I tried to add this in #1287

### CODE REVIEW

File: docker/synapse.Dockerfile [hunk 1]
```diff
@@ -8,7 +8,7 @@ ENV DEBIAN_FRONTEND noninteractive
 
 RUN apt-get -qq update && apt-get -qq install -y \
         apt-utils ${PYTHON_VERSION} ${PYTHON_VERSION}-dev ${PYTHON_VERSION}-venv \
-        python3-pip eatmydata redis-server
+        python3-pip eatmydata redis-server cur
```
Comment (DMRobertson) [inline docker/synapse.Dockerfile:11]: is this meant to be `curl`?

Comment (erikjohnston) [reply docker/synapse.Dockerfile:11]: SIGH, yes thank you. My crankness continues

**Review by DMRobertson — CHANGES_REQUESTED** [review]
Can we add something which would detect the failure to invoke curl? See https://github.com/matrix-org/sytest/pull/1288#issuecomment-1243505026

File: docker/synapse.Dockerfile [hunk 2]
```diff
@@ -6,9 +6,12 @@ ARG PYTHON_VERSION=python3
 
 ENV DEBIAN_FRONTEND noninteractive
 
+# Ensire we die correctly when using pipes in RUN.
```
Comment (DMRobertson) [inline docker/synapse.Dockerfile:9]:
```suggestion
# Ensure we die correctly when using pipes in RUN.
```

Otherwise LGTM, I'm happy if CI is

**Review by DMRobertson — APPROVED** [review]

**Comment by erikjohnston** [comment]
Ermh, not sure why a single Dendrite test is broken, but given nothing in this PR has changed anything to do with them I'm going to merge as is.

**Pull request merged**